src/chunks.jl

```diff
@@ -1,272 +1,295 @@
+Base.size(gc::GridChunks) = map(length, gc.chunks)
+Base.:(==)(gc1::GridChunks, gc2::GridChunks) = gc1.chunks == gc2.chunks
 
-Base.:(==)(g1::GridChunks, g2::GridChunks) = g1.chunks == g2.chunks
+# Utils
 
-function subsetchunks_fallback(r, subs)
-    #This is a fallback method that should work for Regular and Irregular chunks r 
-    #Assuming the desired subset is sorted
-    # We simply compute the chunk for every element in subs and collect everything together 
-    #again in either a Regular or IrregularChunk
-    rev = if issorted(subs)
-        false
-    elseif issorted(subs; rev=true)
-        true
-    else
-        rle = chunk_rle(r, subs)
-        return chunktype_from_chunksizes(rle)
-    end
-    cs = zeros(Int, length(r))
-    for i in subs
-        cs[findchunk(r, i)] += 1
-    end
-    # Find first and last chunk where elements are extracted
-    i1 = findfirst(!iszero, cs)
-    i2 = findlast(!iszero, cs)
-    if isnothing(i1) || isnothing(i2)
-        error("Should not be reached. Non-zero indices not found")
-    else
-        chunks = cs[i1:i2]
-        if rev
-            reverse!(chunks)
+# Computes the run length of which chunk is accessed how many times consecutively.
+function chunk_runlength(chunks::ChunkVector, vec)
+    out = Int[]
+    currentchunk = -1
+    for i in vec
+        nextchunk = findchunk(chunks, i)
+        if nextchunk == currentchunk
+            out[end] += 1
+        else
+            push!(out, 1)
+            currentchunk = nextchunk
         end
-        return chunktype_from_chunksizes(chunks)
     end
+    return out
 end
 
-"""
-    chunktype_from_chunksizes(chunks)
-
-Utility function that constructs either a `RegularChunks` or an
-`IrregularChunks` object based on a vector of chunk sizes given as worted Integers. Wherever
-possible it will try to create a regular chunks object.  
-"""
-function chunktype_from_chunksizes(chunks)
-    if length(chunks) == 1
-        #only a single chunk is affected
-        return RegularChunks(chunks[1], 0, chunks[1])
-    elseif length(chunks) == 2
-        #Two affected chunks
-        chunksize = max(chunks[1], chunks[2])
-        return RegularChunks(chunksize, chunksize - chunks[1], sum(chunks))
-    elseif all(==(chunks[2]), view(chunks, (2):(length(chunks) - 1))) &&
-        chunks[end] <= chunks[2] &&
-        chunks[1] <= chunks[2]
-        #All chunks have the same size, only first and last chunk can be shorter
-        chunksize = chunks[2]
-        return RegularChunks(chunksize, chunksize - chunks[1], sum(chunks))
+# Utility function that constructs either a `RegularChunks` or an
+# `IrregularChunks` object based on a vector of chunk sizes given as worted Integers. Wherever
+# possible it will try to create a regular chunks object.  
+function chunktype_from_chunksizes(chunksizes::AbstractVector)
+    if length(chunksizes) == 1
+        # only a single chunk is affected
+        return RegularChunks(chunksizes[1], 0, chunksizes[1])
```

Comment:
Should be fixed here or in a separate PR: fix the case where `chunksizes[1] == 0` maybe this is even the culprit of some of the errors you are seeing.

Reply:
Maybe separate PR I'm trying not to change any logic here, it's just renames etc